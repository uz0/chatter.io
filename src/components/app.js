import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Router, Route, browserHistory, IndexRedirect } from 'react-router';

import Layout from '@/components/layout';

import SignIn from '@/components/sign-in_container';
import SignUp from '@/components/sign-up_container';
import ForgotPassword from '@/components/forgot-password_container';
import Chat from '@/components/chat_container';

import store from '@/store';

export default class App extends Component {
  render() {
    return <Provider store={store}>
      <Router history={browserHistory}>
        <Route path="/" component={Layout}>
          <IndexRedirect to="chat" />
          <Route component={Chat} path="chat" />
          <Route component={SignIn} path="sign-in" />
          <Route component={SignIn} path="sign-in/:code" />
          <Route component={SignUp} path="sign-up" />
          <Route component={SignUp} path="sign-up/:code" />
          <Route component={ForgotPassword} path="/forgot-password" />
        </Route>
      </Router>
    </Provider>;
  }
}
