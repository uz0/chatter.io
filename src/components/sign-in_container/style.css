.login {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  min-height: 0;
  padding-left: var(--layout-padding);
  padding-right: var(--layout-padding);
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #122033;
}

.login .logo {
  font-family: var(--font-family-title);
  font-size: 30px;
  font-weight: var(--font-weight-bold);
  color: #fff;
}

.login nav {
  display: flex;
  justify-content: space-between;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
}

.login nav a {
  --height: 40px;
  position: relative;
  height: var(--height);
  line-height: var(--height);
  font-size: 16px;
  font-weight: var(--font-weight-bold);
  color: var(--color);
  transition-property: color;
}

.login nav a::before {
  content: "";
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  height: 1px;
  background-color: #fff;
  opacity: var(--line-opacity);
  transition-property: opacity;
}

.login nav a:nth-of-type(n+2) {
  margin-left: 20px;
}

.login nav a:not(.active_link) {
  --color: #a0a7b0;
  --line-opacity: 0;
}

.login nav a.active_link {
  --color: #fff;
  --line-opacity: 1;
}

.login .form {
  width: 400px;
  margin-top: 34px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 700px) {
  .login .form {
    width: 100%;
  }
}

