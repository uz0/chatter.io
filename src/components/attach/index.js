import React, { Component, Fragment } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import find from 'lodash/find';
import findIndex from 'lodash/findIndex';
import filter from 'lodash/filter';
import CRC32 from 'crc-32';
import Recorder from 'opus-recorder';
import { uid } from '@/helpers';
import { api } from '@';
import { actions as notificationActions } from '@/components/notification';
import inputActions from '@/components/messages_container/input/actions';
import style from './style.css';

const bytesSize = 1048576;

export const recordStatuses = {
  IDDLE: 'iddle',
  TRANSCRIPT: 'transcript',
  RECORD: 'record',
  DISABLED: 'disabled',
};

const RECORD_NAME = '__record';

function captureMicrophone(errorCallback) {
  return navigator.mediaDevices
    .getUserMedia({ audio: true })
    .catch(errorCallback);
}

class Attach extends Component {
  state = {
    attachments: this.props.defaultValue ? this.props.defaultValue : [],
    recordStatus: recordStatuses.IDDLE,
  };

  getTranscript = async upload_id => {
    const wait = time => new Promise(resolve => setTimeout(() => resolve(), time));
    const {result: {status, transcript}} = await api.voiceRecognitionResult({ upload_id });

    if (status === 'completed') {
      return transcript;
    }

    if (status === 'failed') {
      return null;
    }

    await wait(500);
    return this.getTranscript(upload_id);
  };

  startRecord = async () => {
    if (!Recorder.isRecordingSupported()) {
      console.log('Recording is not support');
      return;
    }

    const microphone = await captureMicrophone(() => {
      this.setState({ recordStatus: recordStatuses.DISABLED });

      this.props.showNotification({
        type: 'error',
        text: 'Unable to access your microphone',
      });
    });

    if (!microphone) {
      return;
    }

    console.log('microphone', microphone);

    const config = {
      monitorGain: 0,
      recordingGain: 1,
      numberOfChannels: 1,
      encoderSampleRate: 24000,
      originalSampleRateOverride: 16000,
      encoderPath: 'assets/encoderWorker.min.js',
    };

    this.recorder = new Recorder(config);
    this.recorder.start();
    this.microphone = microphone;
    this.setState({ recordStatus: recordStatuses.RECORD });

    this.timeout = setTimeout(() => {
      if (this.state.recordStatus === recordStatuses.RECORD) {
        this.stopRecord();
      }
    }, 60000);

    this.recorder.onstart = () => {
      console.log('start record');
    };

    this.recorder.onstop = () => {
      console.log('stop record');
    };

    this.recorder.ondataavailable = async typedArray => {
      const dataBlob = new Blob([typedArray], { type: 'audio/ogg' });
      dataBlob.name = RECORD_NAME;

      const requestData = new FormData();
      requestData.append('file', dataBlob);
      requestData.append('encoding', 'OGG_OPUS');
      requestData.append('rate_hertz', 24000);
      requestData.append('audio_channels', 1);

      const attachment = {
        uid: uid(),
        byte_size: dataBlob.size,
        content_type: dataBlob.type,
        file_name: RECORD_NAME,
        preview: '',
        url: '',
        upload_id: null,
        isLoading: true,
      };

      this.loadFileByChunks(dataBlob, attachment.uid);
      this.setState({ attachments: [attachment] });
      console.log('this.recorder.blob', attachment);
    };
  };

  stopRecord = async () => {
    this.recorder.stop();

    if (this.timeout) {
      clearTimeout(this.timeout);
    }
  };

  resetAttachment = () => {
    const input = document.getElementById(this.props.uniqueId);
    input.value = '';
    this.setState({ attachments: [] });
  };

  getBlobBase = blob => new Promise(resolve => {
    const reader = new FileReader();

    reader.onloadend = event => {
      let binaryString = '';
      const bytes = new Uint8Array(event.target.result);

      for (let i = 0; i < bytes.byteLength; i++) {
        binaryString += String.fromCharCode(bytes[i]);
      }

      resolve(window.btoa(binaryString));
    };

    reader.readAsArrayBuffer(blob);
  });

  getFileChecksum = file => new Promise(resolve => {
    const reader = new FileReader();

    reader.onloadend = event => {
      const bytes = new Uint8Array(event.target.result);
      resolve(CRC32.buf(bytes) >>> 0);
    };

    reader.readAsArrayBuffer(file);
  });

  onAttachFileChange = event => this.attachFiles(event.target.files);

  attachFiles = files => {
    let isSizeValid = true;

    [].forEach.call(files, file => {
      if (file.size > 209715200) {
        isSizeValid = false;
      }
    });

    if (!isSizeValid) {
      this.props.showNotification({
        type: 'info',

        text: this.props.t(
          'validation_max_size',

          {
            object: this.props.t('file'),
            count: 200,
            size_type: this.props.t('mb').toLowerCase(),
          },
        ),
      });

      return;
    }

    let attachments = [];

    [].forEach.call(files, file => {
      attachments.push({
        uid: uid(),
        byte_size: file.size,
        content_type: file.type,
        file_name: file.name,
        preview: '',
        url: '',
        upload_id: null,
        isLoading: true,
      });
    });

    this.setState({ attachments });

    [].forEach.call(files, (file, index) => {
      const attachment = attachments[index];
      setTimeout(() => this.loadFileByChunks(file, attachment.uid));
      const reader = new FileReader();

      reader.onloadend = () => {
        attachments[index].preview = reader.result;
        attachments[index].url = reader.result;
        this.setState({ attachments });
      };

      reader.readAsDataURL(file);
    });

    const input = document.getElementById(this.props.uniqueId);
    input.value = '';
  };

  updateAttachmentState = (uid, data) => {
    let attachments = this.state.attachments;
    const index = findIndex(attachments, { uid });

    if (index === -1) {
      return;
    }

    attachments[index] = {
      ...attachments[index],
      ...data,
    };

    let isLoaded = true;

    attachments.forEach(attachment => {
      if (attachment.byte_size !== attachment.currentChunk) {
        isLoaded = false;
      }
    });

    if (this.props.onChange && isLoaded) {
      this.props.onChange(attachments);
    }

    this.setState({ attachments });
  };

  loadFileByChunks = async (file, uid) => {
    try {
      if (file.size <= bytesSize) {
        await this.loadFullFile(file, uid);
        return;
      }

      await this.loadMainPartFile(file, uid);
    } catch (error) {
      console.error(error);

      this.props.showNotification({
        type: 'error',
        text: error.text,
      });

      this.resetAttachment();
    }
  };

  loadFullFile = async (file, uid) => {
    const chunk = await this.getBlobBase(file);
    const checksum = await this.getFileChecksum(file);

    let object = {
      chunk_id: 0,
      async: true,
      file_chunk: chunk,
      upload_id: null,
      file_size: file.size,
      file_checksum: checksum,
      file_name: file.name,
      content_type: file.type,
    };
    
    if (file.type === 'audio/ogg') {
      object.voice_recognition = true;
      object.subscription_id = this.props.subscription_id;
    }

    const response = await api.attachmentByChunks({
      chunk_id: 0,
      async: true,
      file_chunk: chunk,
      upload_id: null,
      file_size: file.size,
      file_checksum: checksum,
      file_name: file.name,
      content_type: file.type,
    });

    if (file.type === 'audio/ogg') {
      this.setState({ recordStatus: recordStatuses.TRANSCRIPT });
      await api.voiceRecognition({ upload_id: response.upload_id });
      const transcript = await this.getTranscript(response.upload_id);
      this.setState({recordStatus: recordStatuses.IDDLE});

      if (transcript) {
        this.props.setText(transcript);
      }
    }

    this.updateAttachmentState(uid, {
      currentChunk: file.size,
      isLoading: false,
      upload_id: response.upload_id,
    });
  };

  loadMainPartFile = async (file, uid) => {
    let attachment = find(this.state.attachments, { uid });
    let blob = file.slice(0, bytesSize);
    let chunk = await this.getBlobBase(blob);
    let chunk_id = 1;
    const checksum = await this.getFileChecksum(file);

    const response = await api.attachmentByChunks({
      chunk_id: 0,
      async: true,
      file_chunk: chunk,
      upload_id: null,
      file_size: file.size,
      file_checksum: checksum,
      file_name: file.name,
    });

    attachment.currentChunk = bytesSize;
    attachment.upload_id = response.upload_id;

    if (file.size <= bytesSize) {
      attachment.isLoading = false;
    }

    this.updateAttachmentState(uid, attachment);

    if (file.size - bytesSize < bytesSize) {
      this.loadLastChunk(file, uid);
      return;
    }

    for (let i = bytesSize; i <= file.size; i += bytesSize, chunk_id++) {
      const currentStateAttachment = find(this.state.attachments, { uid });

      if (!currentStateAttachment) {
        break;
      }

      attachment.currentChunk = i;
      this.updateAttachmentState(uid, attachment);

      blob = file.slice(i, i + bytesSize);
      chunk = await this.getBlobBase(blob);

      await api.attachmentByChunks({
        chunk_id,
        async: true,
        file_chunk: chunk,
        upload_id: response.upload_id,
        file_size: file.size,
        file_checksum: checksum,
        file_name: file.name,
      });
    }

    const rest = file.size - attachment.currentChunk + bytesSize;

    if (rest > 0 && rest < bytesSize) {
      this.loadLastChunk(file, uid, chunk_id + 1);
      return;
    }

    attachment.currentChunk = file.size;
    attachment.isLoading = false;
    this.updateAttachmentState(uid, attachment);
  };

  loadLastChunk = async (file, uid, chunk_id) => {
    const currentStateAttachment = find(this.state.attachments, { uid });

    if (!currentStateAttachment) {
      return;
    }

    const blob = file.slice(currentStateAttachment.currentChunk, file.size);
    const chunk = await this.getBlobBase(blob);
    const checksum = await this.getFileChecksum(file);

    await api.attachmentByChunks({
      chunk_id,
      async: true,
      file_chunk: chunk,
      upload_id: currentStateAttachment.upload_id,
      file_size: file.size,
      file_checksum: checksum,
      file_name: file.name,
    });

    this.updateAttachmentState(uid, {
      currentChunk: file.size,
      isLoading: false,
    });
  };

  removeAttachment = uid => () => {
    let attachments = [...this.state.attachments];
    const index = findIndex(attachments, { uid });

    if (index !== -1) {
      attachments.splice(index, 1);
    }

    if (this.props.onChange) {
      this.props.onChange(attachments);
    }

    this.setState({ attachments });
  };

  async componentWillMount() {
    await captureMicrophone(() => {
      this.setState({ recordStatus: recordStatuses.DISABLED });
    });
  }

  render() {
    const images = this.state.attachments && filter(this.state.attachments, attachment => attachment.content_type.match('image/')) || [];
    const files = this.state.attachments && filter(this.state.attachments, attachment => !attachment.content_type.match('image/')) || [];

    return <Fragment>
      <input
        id={this.props.uniqueId}
        className={style.attach_input}
        type="file"
        ref={node => this.attachInputRef = node}
        onChange={this.onAttachFileChange}
        multiple
      />

      {this.props.children && this.props.children({
        files,
        images,
        removeAttachment: this.removeAttachment,
        startRecord: this.startRecord,
        stopRecord: this.stopRecord,
        recordStatus: this.state.recordStatus,
      })}
    </Fragment>;
  }
}

export default compose(
  withTranslation(),

  connect(
    null,

    {
      setText: inputActions.setText,
      showNotification: notificationActions.showNotification,
    },
  ),
)(Attach);
