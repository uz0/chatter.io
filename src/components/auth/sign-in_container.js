import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import get from 'lodash/get';
import Link from '@/components/link';
import Button from '@/components/button';
import Loading from '@/components/loading';
import Validators from '@/components/form/validators';
import Form from '@/components/new-form/form';
import Input from '@/components/input';
import { withTranslation } from 'react-i18next';
import { actions as notificationActions } from '@/components/notification';
import { actions as formActions } from '@/components/form';
import { actions as storeActions } from '@/store';
import { api } from '@';
import { withRouter } from '@/hoc';
import style from './style.css';

class SignIn extends Component {
  state = {
    commonError: '',
    isLoading: false,
  };

  onSubmit = async (data, onError) => {
    this.setState({ isLoading: true });

    try {
      const { user } = await api.login({
        email: data.email.value,
        password: data.password.value,
      });

      this.setState({ isLoading: false });
      this.props.formReset('login');
      this.props.setCurrentUser(user);
      window.localStorage.setItem('authToken', user.auth_token);
      window.localStorage.setItem('currentUser', JSON.stringify(user));

      this.props.showNotification({
        type: 'success',
        text: this.props.t('you_logged_success'),
      });

      this.props.pushUrl('/chat');
    } catch (error) {
      console.error(error);
      this.setState({ isLoading: false });
      onError(error.text);
    }
  };

  componentWillMount() {
    if (this.props.currentUser) {
      this.props.pushUrl('/');
      return;
    }
  }

  render() {
    const fields = {
      email: {
        validations: [
          {
            action: Validators.required,
            text: this.props.t('validation_required', { field: this.props.t('email') }),
          },

          {
            action: Validators.email,
            text: this.props.t('validation_correct', { field: this.props.t('email') }),
          },
        ],
      },

      password: {
        validations: [
          {
            action: Validators.required,
            text: this.props.t('validation_required', { field: this.props.t('password') }),
          },

          {
            action: Validators.minLength(6),
            text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
          },

          {
            action: Validators.contains(' '),
            text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
          },
        ],
      },
    };

    return <div className={style.auth}>
      {this.state.isLoading &&
        <Loading type="line" isShown className={style.loading} />
      }

      <div className={style.header}>
        <h1 className={style.title}>Unichat</h1>
        <Link to="/sign-up" className={style.link}>Sign Up</Link>
      </div>

      <div className={style.content}>
        <h3 className={style.subtitle}>Welcome back!</h3>

        <Form
          fields={fields}
          onSubmit={this.onSubmit}
          isLoading={this.state.isLoading}
          className={style.form}
        >
          {({ getInputProps, submitProps }) => {
            return <Fragment>
              <Input
                {...getInputProps('email')}
                appearance="_none-classic"
                className={style.input}
                placeholder={this.props.t('email')}
              />

              <Input
                {...getInputProps('password')}
                type="password"
                appearance="_none-classic"
                className={style.input}
                placeholder={this.props.t('password')}
              />

              <div className={style.actions}>
                <Button
                  appearance="_basic-primary"
                  text="Log in"
                  className={style.submit}
                  {...submitProps}
                />

                <Link to="/forgot-password" className={style.forgot}>Forgot password?</Link>
              </div>
            </Fragment>;
          }}
        </Form>
      </div>

      <p className={style.copyrights}>Â© Unichat (Beta 1.3)</p>
    </div>;
  }
}

export default compose(
  withRouter,
  withTranslation(),

  connect(
    state => ({
      currentUser: state.currentUser,
      email: get(state.forms, 'login.email', {}),
      password: get(state.forms, 'login.password', {}),
    }),

    {
      formReset: formActions.formReset,
      showNotification: notificationActions.showNotification,
      setCurrentUser: storeActions.setCurrentUser,
    },
  ),
)(SignIn);
