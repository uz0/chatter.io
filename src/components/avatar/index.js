import React from 'react';
import classnames from 'classnames/bind';
import style from './style.css';

const cx = classnames.bind(style);

const Avatar = ({ className, letter, photo, isGroup }) => <div
  className={cx('avatar', className, {'_is-group': isGroup})}
  {...photo ? { style: {'--photo': `url(${photo})`} } : {}}
>
  {letter && letter}
</div>;

export default Avatar;
