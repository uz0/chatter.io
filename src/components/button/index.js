import React, { Component } from 'react';
import classnames from 'classnames/bind';
import Icon from '@/components/icon';
import Loading from '@/components/loading';
import style from './style.css';

const cx = classnames.bind(style);

class Button extends Component {
  render = () => <button
    type={this.props.type}
    ref={this.props.ref}
    className={cx('button', this.props.className)}
    appearance={this.props.appearance}
    onClick={this.props.onClick}
    {...this.props.disabled ? {disabled: true} : {}}
    id={this.props.id}
  >
    {this.props.icon && <Icon name={this.props.icon} />}
    {this.props.text && <span>{ this.props.text }</span>}

    {this.props.isLoading &&
      <Loading isShown className={style.loading} />
    }
  </button>;
}

export default Button;
