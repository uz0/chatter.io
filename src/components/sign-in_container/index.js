import React, { Component } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import Link from '@/components/link';
import Form from '@/components/form/form';
import Validators from '@/components/form/validators';
import Input from '@/components/form/input';
import { withNamespaces } from 'react-i18next';
import { actions as notificationActions } from '@/components/notification';
import { actions as formActions } from '@/components/form';
import { actions as storeActions } from '@/store';
import { api } from '@';
import { withRouter } from '@/hoc';
import style from './style.css';

class SignIn extends Component {
  state = {
    commonError: {},
    isLoading: false,
  };

  submit = event => {
    event.preventDefault();
    this.setState({ isLoading: true });

    api.login({
      email: this.props.formData.email.value,
      password: this.props.formData.password.value,
    }).then(data => {
      console.log('data', data);
      this.setState({ isLoading: false });
      this.props.formReset('login');
      this.props.setCurrentUser(data.user);
      window.localStorage.setItem('authToken', data.user.auth_token);
      window.localStorage.setItem('currentUser', JSON.stringify(data.user));
      this.props.showNotification(this.props.t('you_logged_success'));
      this.props.pushUrl('/chat');
    }).catch(error => {
      this.setState({ commonError: this.props.t(error.code), isLoading: false });
      this.props.showNotification(this.props.t(error.code));
    });
  }

  componentWillMount() {
    if (this.props.currentUser) {
      this.props.pushUrl('/');
      return;
    }
  }

  render = () => {
    return <div className={style.login}>
      <Link to="/" className={style.logo}>Unichat</Link>

      <nav>
        <Link className={style.active_link} to="/sign-in">{this.props.t('log_in')}</Link>
        <Link to="/sign-up">{this.props.t('sign_up')}</Link>
      </nav>

      <Form
        actions={[
          { text: `${this.props.t('log_in')}`, appearance: '_basic-primary', onClick: this.submit, type: 'submit' },
          { text: `${this.props.t('forgot_password')}?`, to: '/forgot-password' },
        ]}

        error={this.state.commonError.text || ''}
        model="login"
        disabled={this.state.isLoading}
        className={style.form}
      >
        <Input
          type="email"
          class="light"
          placeholder={this.props.t('email')}
          model="login.email"
          disabled={this.state.isLoading}
          className={style.input}

          validations={[
            {
              action: Validators.required,
              text: this.props.t('validation_required', { field: this.props.t('email') }),
            },

            {
              action: Validators.email,
              text: this.props.t('validation_correct', { field: this.props.t('email') }),
            },
          ]}
        />

        <Input
          type="password"
          class="light"
          placeholder={this.props.t('password')}
          model="login.password"
          disabled={this.state.isLoading}
          className={style.input}

          validations={[
            {
              action: Validators.required,
              text: this.props.t('validation_required', { field: this.props.t('password') }),
            },

            {
              action: Validators.minLength(6),
              text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
            },

            {
              action: Validators.contains(' '),
              text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
            },
          ]}
        />
      </Form>
    </div>;
  }
}

export default compose(
  withRouter,
  withNamespaces('translation'),

  connect(
    state => ({
      formData: state.forms.login,
      currentUser: state.currentUser,
    }),

    {
      formReset: formActions.formReset,
      showNotification: notificationActions.showNotification,
      setCurrentUser: storeActions.setCurrentUser,
    },
  ),
)(SignIn);
