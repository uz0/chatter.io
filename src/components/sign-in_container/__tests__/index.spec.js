import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from '@/components/sign-in_container';

describe('SignIn component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SignIn
      t={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
