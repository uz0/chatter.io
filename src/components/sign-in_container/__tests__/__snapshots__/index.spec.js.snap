// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(CustomLink)
          className="logo"
          to="/"
        >
          Unichat
        </withRouter(CustomLink)>,
        <nav>
          <withRouter(CustomLink)
            className="active_link"
            to="/sign-in"
          />
          <withRouter(CustomLink)
            to="/sign-up"
          />
        </nav>,
        <Connect(Form)
          actions={
            Array [
              Object {
                "appearance": "_basic-primary",
                "id": "submit",
                "onClick": [Function],
                "text": "undefined",
                "type": "submit",
              },
              Object {
                "text": "undefined?",
                "to": "/forgot-password",
              },
            ]
          }
          className="form"
          disabled={false}
          error=""
          model="login"
        >
          <Connect(Input)
            className="input"
            disabled={false}
            id="email"
            model="login.email"
            type="email"
            validations={
              Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ]
            }
          />
          <Connect(Input)
            className="input"
            disabled={false}
            id="password"
            model="login.password"
            type="password"
            validations={
              Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ]
            }
          />
        </Connect(Form)>,
      ],
      "className": "login",
      "data-6f1ad961": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Unichat",
          "className": "logo",
          "to": "/",
        },
        "ref": null,
        "rendered": "Unichat",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(CustomLink)
              className="active_link"
              to="/sign-in"
            />,
            <withRouter(CustomLink)
              to="/sign-up"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "active_link",
              "to": "/sign-in",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "to": "/sign-up",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            Object {
              "appearance": "_basic-primary",
              "id": "submit",
              "onClick": [Function],
              "text": "undefined",
              "type": "submit",
            },
            Object {
              "text": "undefined?",
              "to": "/forgot-password",
            },
          ],
          "children": Array [
            <Connect(Input)
              className="input"
              disabled={false}
              id="email"
              model="login.email"
              type="email"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />,
            <Connect(Input)
              className="input"
              disabled={false}
              id="password"
              model="login.password"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />,
          ],
          "className": "form",
          "disabled": false,
          "error": "",
          "model": "login",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "input",
              "disabled": false,
              "id": "email",
              "model": "login.email",
              "placeholder": undefined,
              "type": "email",
              "validations": Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "input",
              "disabled": false,
              "id": "password",
              "model": "login.password",
              "placeholder": undefined,
              "type": "password",
              "validations": Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(CustomLink)
            className="logo"
            to="/"
          >
            Unichat
          </withRouter(CustomLink)>,
          <nav>
            <withRouter(CustomLink)
              className="active_link"
              to="/sign-in"
            />
            <withRouter(CustomLink)
              to="/sign-up"
            />
          </nav>,
          <Connect(Form)
            actions={
              Array [
                Object {
                  "appearance": "_basic-primary",
                  "id": "submit",
                  "onClick": [Function],
                  "text": "undefined",
                  "type": "submit",
                },
                Object {
                  "text": "undefined?",
                  "to": "/forgot-password",
                },
              ]
            }
            className="form"
            disabled={false}
            error=""
            model="login"
          >
            <Connect(Input)
              className="input"
              disabled={false}
              id="email"
              model="login.email"
              type="email"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />
            <Connect(Input)
              className="input"
              disabled={false}
              id="password"
              model="login.password"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />
          </Connect(Form)>,
        ],
        "className": "login",
        "data-6f1ad961": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Unichat",
            "className": "logo",
            "to": "/",
          },
          "ref": null,
          "rendered": "Unichat",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(CustomLink)
                className="active_link"
                to="/sign-in"
              />,
              <withRouter(CustomLink)
                to="/sign-up"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "className": "active_link",
                "to": "/sign-in",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "to": "/sign-up",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actions": Array [
              Object {
                "appearance": "_basic-primary",
                "id": "submit",
                "onClick": [Function],
                "text": "undefined",
                "type": "submit",
              },
              Object {
                "text": "undefined?",
                "to": "/forgot-password",
              },
            ],
            "children": Array [
              <Connect(Input)
                className="input"
                disabled={false}
                id="email"
                model="login.email"
                type="email"
                validations={
                  Array [
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                  ]
                }
              />,
              <Connect(Input)
                className="input"
                disabled={false}
                id="password"
                model="login.password"
                type="password"
                validations={
                  Array [
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                  ]
                }
              />,
            ],
            "className": "form",
            "disabled": false,
            "error": "",
            "model": "login",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "disabled": false,
                "id": "email",
                "model": "login.email",
                "placeholder": undefined,
                "type": "email",
                "validations": Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "disabled": false,
                "id": "password",
                "model": "login.password",
                "placeholder": undefined,
                "type": "password",
                "validations": Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
